packages/core/data/oot/world/shadow_temple.yml

```diff
@@ -44,75 +45,94 @@
+    "Shadow Temple GS Invisible Scythe": "soul_like_like && soul_keese && (has_ranged_weapon_adult || can_use_slingshot || can_use_sticks || has_explosives || has_weapon) && (can_collect_distance || climb_anywhere)"
     "Shadow Temple SR Pit 1": "true"
     "Shadow Temple SR Pit 2": "true"
     "Shadow Temple SR Pit 3": "true"
     "Shadow Temple SR Pit 4": "true"
     "Shadow Temple SR Pit 5": "true"
 "Shadow Temple Falling Spikes":
   dungeon: Shadow
-  exits:
-    "Shadow Temple Open": "true"
   locations:
     "Shadow Temple Falling Spikes Lower": "true"
-    "Shadow Temple Falling Spikes Upper 1": "is_adult && has(STRENGTH) && has_lens"
-    "Shadow Temple Falling Spikes Upper 2": "is_adult && has(STRENGTH) && has_lens"
-    "Shadow Temple GS Falling Spikes": "can_collect_distance"
+    "Shadow Temple Falling Spikes Upper 1": "(is_adult && has(STRENGTH)) || climb_anywhere || hookshot_anywhere"
+    "Shadow Temple Falling Spikes Upper 2": "(is_adult && has(STRENGTH)) || climb_anywhere || hookshot_anywhere"
+    "Shadow Temple GS Falling Spikes": "can_collect_distance || (climb_anywhere && can_damage_skull)"
     "Shadow Temple Pot Falling Spikes Bottom 1": "true"
     "Shadow Temple Pot Falling Spikes Bottom 2": "true"
-    "Shadow Temple Pot Falling Spikes Top 1": "is_adult && has(STRENGTH) && has_lens"
-    "Shadow Temple Pot Falling Spikes Top 2": "is_adult && has(STRENGTH) && has_lens"
+    "Shadow Temple Pot Falling Spikes Top 1": "(is_adult && has(STRENGTH)) || climb_anywhere || hookshot_anywhere"
+    "Shadow Temple Pot Falling Spikes Top 2": "(is_adult && has(STRENGTH)) || climb_anywhere || hookshot_anywhere"
 "Shadow Temple Invisible Spikes":
   dungeon: Shadow
   exits:
-    "Shadow Temple Open": "true"
-    "Shadow Temple Wind": "small_keys_shadow(3) && can_hookshot"
+    "Shadow Temple Open": "small_keys_shadow(4)"
+    "Shadow Temple Wind Front": "cond(climb_anywhere, small_keys_shadow(4), small_keys_shadow(3) && (is_adult && can_hookshot) || can_longshot)"
     "Shadow Temple Skull Pot": "has(RUPEE_SILVER_SHADOW_SPIKES, 5)"
   locations:
-    "Shadow Temple SR Spikes Back Left": "has_lens && can_hookshot"
-    "Shadow Temple SR Spikes Right": "can_hookshot"
+    "Shadow Temple SR Spikes Back Left": "has_lens && (can_hookshot || climb_anywhere)"
+    "Shadow Temple SR Spikes Right": "can_hookshot || climb_anywhere"
     "Shadow Temple SR Spikes Center": "true"
-    "Shadow Temple SR Spikes Front Left": "can_hookshot"
-    "Shadow Temple SR Spikes Midair": "has_lens && can_hookshot"
-    "Shadow Temple Invisible Spike Room": "soul_redead_gibdo && (has_weapon || can_use_sticks || can_use_din)"
+    "Shadow Temple SR Spikes Front Left": "can_hookshot || climb_anywhere"
+    "Shadow Temple SR Spikes Midair": "has_lens && (can_hookshot || climb_anywhere) && (has_weapon || can_use_sticks || can_hammer || has_hover_boots)"
```

Comment:
Can't you just roll jump to it?

Reply:
nope, too far.

Reply:
I am reminded, though, that hookshot anywhere as adult or longshot anywhere as child should be able to reach it.